database/migrations/2021_06_01_181900_refactor_timestamps_anew.php

```diff
@@ -0,0 +1,563 @@
+			$table->renameColumn(self::UPDATED_AT_COL_NAME, self::UPDATED_AT_COL_NAME . '_tmp');
+		});
+		// Create the new columns as temporarily nullable
+		Schema::table($tableName, function (Blueprint $table) {
+			$table->dateTime(
+				self::CREATED_AT_COL_NAME,
+				self::DATETIME_PRECISION
+			)->nullable();
+			$table->dateTime(
+				self::UPDATED_AT_COL_NAME,
+				self::DATETIME_PRECISION
+			)->nullable();
+		});
+		$needsConversion = $this->needsConversionOnUpgrade();
+		DB::beginTransaction();
+		$entities = DB::table($tableName)->select([
+			self::ID_COL_NAME,
+			self::CREATED_AT_COL_NAME . '_tmp',
+			self::UPDATED_AT_COL_NAME . '_tmp',
+		])->lazyById();
+		foreach ($entities as $entity) {
+			$created_at = $entity->{self::CREATED_AT_COL_NAME . '_tmp'};
+			$updated_at = $entity->{self::UPDATED_AT_COL_NAME . '_tmp'};
+			if ($needsConversion) {
+				$created_at = $this->upgradeDatetime($created_at);
+				$updated_at = $this->upgradeDatetime($updated_at);
+			}
+			DB::table($tableName)->where(self::ID_COL_NAME, '=', $entity->id)->update([
+				self::CREATED_AT_COL_NAME => $created_at,
+				self::UPDATED_AT_COL_NAME => $updated_at,
+			]);
+		}
+		DB::commit();
+		// Make the new columns non-nullable
+		Schema::table($tableName, function (Blueprint $table) {
+			$table->dateTime(
+				self::CREATED_AT_COL_NAME,
+				self::DATETIME_PRECISION
+			)->nullable(false)->change();
+			$table->dateTime(
+				self::UPDATED_AT_COL_NAME,
+				self::DATETIME_PRECISION
+			)->nullable(false)->change();
+		});
+		// We must use two single calls to work around an SQLite limitation
+		Schema::table($tableName, function (Blueprint $table) {
+			$table->dropColumn(self::CREATED_AT_COL_NAME . '_tmp');
+		});
+		Schema::table($tableName, function (Blueprint $table) {
+			$table->dropColumn(self::UPDATED_AT_COL_NAME . '_tmp');
+		});
+	}
+
+	/**
+	 * Downgrades the systems times `created_at` and `updated_at` for each
+	 * table in {@link RefactorTimestampsAnew::TABLE_NAMES}.
+	 */
+	protected function downgradeORMSystemTimes(): void
+	{
+		foreach (self::TABLE_NAMES as $tableName) {
+			$this->downgradeORMSystemTimesByTable($tableName);
+		}
+	}
+
+	/**
+	 * Downgrades the system times `created_at` and `updated_at` for the given
+	 * table.
+	 *
+	 * @param string $tableName the name of the table to be downgraded
+	 */
+	protected function downgradeORMSystemTimesByTable(string $tableName): void
+	{
+		// We must use three single calls to work around an SQLite limitation
+		Schema::table($tableName, function (Blueprint $table) {
+			$table->renameColumn(self::CREATED_AT_COL_NAME, self::CREATED_AT_COL_NAME . '_tmp');
+		});
+		Schema::table($tableName, function (Blueprint $table) {
+			$table->renameColumn(self::UPDATED_AT_COL_NAME, self::UPDATED_AT_COL_NAME . '_tmp');
+		});
+		Schema::table($tableName, function (Blueprint $table) {
+			$table->timestamps();
+		});
+		$needsConversion = $this->needsConversionOnUpgrade();
```

Comment:
You are right. That's a copy-&-past error. As discussed on Gitter this accidentally did the right thing, because we must not use a different method during downgrade than during upgrade. Removed `needsConversionOnDowngrade` entirely and renamed `needsConversionOnUpgrade` to `needsConversion`.